src/main/java/pl/javastart/task/phone/Phone.java

```diff
@@ -0,0 +1,37 @@
+package pl.javastart.task.phone;
+
+public class Phone {
+    private Contract contract;
+
+//    public Phone(Contract contract) {
+//        this.contract = contract;
+//    }
```

Comment:
W tej klasie powinien być tylko ten konstruktor. Jeżeli pojawi się inny typ umowy, to z takim konstruktorem będzie można ją wykorzystać, bez wprowadzania żadnych zmian do klasy Phone. W Twoim aktualnym podejściu dla każdego nowego rodzaju umowy trzeba modyfikować klasę Phone